### PR TITLE
fixed bug with chi2 error

### PR DESCRIPTION
**Pull request by jbland64**

### CODE REVIEW

**Comment by jbland64** [comment]
@FilipJankyTE fill free to review 

main change is a bug fix due to lapacke overwritting the matrix  g_coef_meas_w which is then used in calculating chi2

other changes include:
* including the MASK_LIM mask
* altering the README

File: src/rtgsfit.c [hunk 1]
```diff
@@ -71,11 +71,14 @@ void make_basis(
             basis[i_grid + N_GRID] = 0.0;
         }        
     }
+    
     // could use gradient from previous iteration.  should apply mask 
-    gradient_z(flux_norm, &basis[2*N_GRID]);   
+    if (N_PLS == 3)
+    {
+        gradient_z(flux_norm, &basis[2*N_GRID]);   
+    }
```
Comment (FilipJankyTE) [inline src/rtgsfit.c:79]: What should happen if N_PLS is not 3?

File: src/rtgsfit.c [hunk 2]
```diff
@@ -185,17 +189,17 @@ void rtgsfit(
...
-            N_MEAS, coef, 1, 0.0, meas_model, 1);       
+    cblas_dgemv(CblasRowMajor, CblasTrans,  N_COEF, N_MEAS, 1.0, g_coef_meas_w_orig, 
+            N_MEAS, coef, 1, 0.0, meas_model, 1);    
             
     // find error between meas and model
     *error = 0.0;
     for (i_meas=0; i_meas<N_MEAS; i_meas++)
     {
-        *error += (meas_no_coil[i_meas] -  meas_model[i_meas]) \
+        *error = *error + (meas_no_coil[i_meas] -  meas_model[i_meas]) \
```
Comment (FilipJankyTE) [inline src/rtgsfit.c:199]: I believe if you would do (*error) += it could do what you needed

File: README.md [hunk 3]
```diff
@@ -29,13 +31,38 @@ cblas.h
...
+* function names should begin with a verb e.g. 'get_flux_norm' 
+* boolean variables are simply integers in C. Maybe use short or other inbuilt types
+* Have not used size_t or any other non-inbuilt types
+* Boolean names should begin with is_ prefix to differentiate them from other variables
+* Boolean names should be always be in the "positive" affirmation, with 1 being
+true and 0 being false
+* order of variables should go from largest to smallest e.g. GRID_R_MIN
+as if it was sucessive objects in the OOP paradigme.  
+* Green's functions are prefixed with g_
+
```
Comment (FilipJankyTE) [inline README.md:57]: Nice, I really like it! I will start adopting these things in my code.

Comment (jbland64) [reply README.md:57]: yeah please note the code doesnt actually abide by most of this stuff yet ... put here more of as a reminder for me

File: src/rtgsfit.c (hunk 1 above)
Comment (jbland64) [reply src/rtgsfit.c:79]: currently only works if N_PLS is 2 or 3 probably should make it more explicit and have both options

File: src/rtgsfit.c (hunk 2 above)
Comment (jbland64) [reply src/rtgsfit.c:199]: yeah this wasnt actually the issue but your suggestion is nice and explicit

**Pull request merged**